PostHog/Replay/NetworkRequestSample.swift

```diff
@@ -0,0 +1,257 @@
+                requestContentTypeRaw = contentType
+                requestContentType = ContentType(contentType: contentType)
+            }
+
+            // grab request body
+            if let requestData = request.httpBody ?? request.httpBodyStream?.consume() {
+                if let responseContentType, responseContentType == .image {
+                    // don't record response body for image types
+                    let bodyStr = requestData.base64EncodedString(options: .endLineWithLineFeed)
+                    requestBodyLength = bodyStr.count
+                } else if let utfString = String(data: requestData, encoding: String.Encoding.utf8) {
+                    requestBodyStr = utfString
+                    requestBodyLength = utfString.count
+                }
+            }
+        }
+
+        // called on stopLoading (request was cancelled)
+        func stop() {
+            durationMs = relative(getMonotonicTimeInMilliseconds(), to: requestStartTime)
+        }
+
+        // called on didCompleteWithError
+        func complete(response: URLResponse, error: Error?) {
+            let completedTime = getMonotonicTimeInMilliseconds()
+            responseEndTime = completedTime
+            responseStatus = (response as? HTTPURLResponse)?.statusCode
+            responseHeaders = response.normalizedHeaderFields ?? [:]
+            responseError = error?.localizedDescription
+
+            if let contentType = responseHeaders?["content-type"] as? String {
+                let contentType = contentType.components(separatedBy: ";")[0]
+                responseContentTypeRaw = contentType
+                responseContentType = ContentType(contentType: contentType)
+            }
+
+            durationMs = relative(completedTime, to: requestStartTime)
+
+            if let responseData = responseData as? Data {
+                if let responseContentType, responseContentType == .image {
+                    // don't record response body for image types
+                    let bodyStr = responseData.base64EncodedString(options: .endLineWithLineFeed)
+                    responseBodyLength = bodyStr.count
+                } else if let utfString = String(data: responseData, encoding: String.Encoding.utf8) {
+                    responseBodyStr = utfString
+                    responseBodyLength = utfString.count
+                }
+            }
+        }
+
+        // called after startReceivingData when didReceiveData
+        func didReceive(data: Data) {
+            if responseStartTime == nil {
+                responseStartTime = getMonotonicTimeInMilliseconds()
+                responseData = NSMutableData()
+            }
+            responseData?.append(data)
+        }
+
+        // sample was queued upstream - for debug purposes
+        func markProcessed() {
+            isProcessed = true
+        }
+    }
+
+    private func getMonotonicTimeInMilliseconds() -> UInt64 {
```

Comment:
`getMonotonicTimeInMilliseconds` already exists, can we reuse it? See URLSession extension